### PR TITLE
hx -added all things we need to connect book frontend and backend

### PR DESCRIPTION
**Pull request by zihaoxu20010109**
In this PR, I added all necessary things for connection of Book frontend and backend. TODO:TEST QA AND MUTATION

### CODE REVIEW

File: frontend/src/main/components/Books/BookTable.js [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
+
+
     const columns = [
         {
             Header: 'id',
             accessor: 'id', // accessor is the "key" in the data
         },
         {
-            Header: 'Title',
+            Header: 'TITLE',
```
Comment (wq12li) [inline frontend/src/main/components/Books/BookTable.js]: `TITLE`->`Title`

File: frontend/src/main/components/Books/BookTable.js [hunk 2]
```diff
@@ -1,45 +1,45 @@
...
             accessor: 'title',
         },
         {
-            Header: 'Author',
+            Header: 'AUTHOR',
```
Comment (wq12li) [inline frontend/src/main/components/Books/BookTable.js]: `AUTHOR`->`Author`

File: frontend/src/main/pages/Books/BookDetailsPage.js [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
+        }
+      }
+    );
 
   return (
     <BasicLayout>
       <div className="pt-2">
-        <h1>Book Details</h1>
-        <BookTable books={[response.book]} showButtons={false} />
+        <h1>book Details</h1>
```
Comment (wq12li) [inline frontend/src/main/pages/Books/BookDetailsPage.js]: `book`->`Book`

File: frontend/src/tests/components/Books/BookTable.test.js [hunk 4]
```diff
@@ -1,213 +1,150 @@
...
+		const { getByText, getByTestId } = render(
+			<QueryClientProvider client={queryClient}>
+				<MemoryRouter>
+					<BooksTable books={bookFixtures.threeBooks} currentUser={currentUser} />
+				</MemoryRouter>
+			</QueryClientProvider>
+
+		);
+
+		const expectedHeaders = ["id", "TITLE", "AUTHOR","Description"];
```
Comment (wq12li) [inline frontend/src/tests/components/Books/BookTable.test.js]: `"TITLE", "AUTHOR"`->`"Title", "Author"`

**Review by wq12li — COMMENTED** [review]
I've tested this branch on localhost, through the version already pushed to QA, and with local mutation/coverage testing on both frontend and backend: All methods seem to work correctly.

I just have a few suggestions for keeping some formatting consistent, which I've pointed to through these comments. I doubt we'd lose any points even if we didn't change these, but I suppose there's no harm to doing it.

**Comment by nickzahlis** [comment]
lgtm

**Review by khtran1 — CHANGES_REQUESTED** [review]
lgtm, but I second Yiu-Shing's comments; despite not losing any points most likely, it would be best to stay consistent in formatting.

**Review by wq12li — APPROVED** [review]
Looks good now.

**Pull request merged**